### ISSUE TITLE
Next element not annouced when active element is hidden

### ISSUE THREAD

**Issue by Calidus** [nvaccess/nvda#12807]
When the active element gets hidden, NVDA will not announce the next element in Chrome. 

- NVDA does not have this issue with Firefox.
- JAWS does not have this issue with Chrome.

This maybe related to https://github.com/nvaccess/nvda/issues/11980

**Repo:**

JS Fiddle - https://jsfiddle.net/fh7ywocL/8/

**Steps to reproduce:**
1. Open Chrome
2. Go to https://jsfiddle.net/fh7ywocL/8/
3. Focus the Results section/window
4. Press the down arrow to announce the first button
5. Press enter to hide the first button
6. Press the down arrow

**Expected behavior:**
NVDA should announce the second button

**Actual behavior:**
NVDA announces the third button

**System configuration**
NVDA installed/portable/running from source:
Installed

**NVDA version:**
2020.1

**Windows version:**
Microsoft Windows 10 Pro
Version 10.0.19043 Build 19043

Name and version of other software in use when reproducing the issue:
Latest version of Chrome

Other information about your system:
Other questions
Does the issue still occur after restarting your computer?
Yes

Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, 2020.4 also has this issue

If addons are disabled, is your problem still occurring?
Yes

Did you try to run the COM registry fixing tool in NVDA menu / tools?
No

**Comment by feerrenrut** [comment]
Please specify if you are in browse or focus mode when reproducing. The two modes operate differently and this will help to clear up any misconceptions.

**Comment by feerrenrut** [comment]
This looks like a duplicate of https://github.com/nvaccess/nvda/issues/6606
I'll leave this open because this seems to describe a different setup.

**Comment by Adriani90** [comment]
This is still reproducible in Firefox 111 with NVDA last alpha even after #14611 has been merged. In Crhomium this works as expected now.

cc: @jcsteh

**Comment by Calidus** [comment]
> This is still reproducible in Firefox 111 with NVDA last alpha even after #14611 has been merged. In Crhomium this works as expected now.
> 
> cc: @jcsteh

When you say Chromium works as expected now, is that the NVDA last alpha or 2023.1?

**Comment by Adriani90** [comment]
Tested with last alpha.Von meinem iPhone gesendetAm 14.04.2023 um 14:06 schrieb Zach Whitesel ***@***.***>:﻿

This is still reproducible in Firefox 111 with NVDA last alpha even after #14611 has been merged. In Crhomium this works as expected now.
cc: @jcsteh

When you say Chromium works as expected now, is that the NVDA last alpha or 2023.1?

**Comment by jcsteh** [comment]
The expected behaviour is not correct here. When you remove the first button, NVDA has to place the cursor somewhere - it can't be nowhere - so it places it as near as it can, which is now the second button. It can't remain on the first button because the first button no longer exists. Since the cursor is on the second button (which you can confirm by reading the current line), pressing down arrow will take you to the third button. This is consistent across both browsers.

**Comment by jcsteh** [comment]
Focus is actually redirected to the document, not the second button. For me, this fact was reported inconsistently in both browsers. I think there is a race condition here where pressing the first button focuses it, then it gets clicked, then focus goes to the document. It's possible that NVDA receives the focus event for the first button after the button is already destroyed, in which case things get a bit weird.
